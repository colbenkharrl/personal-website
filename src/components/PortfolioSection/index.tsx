import { FC, ReactNode } from 'react';
import { IconComponent } from '../Icons';

export type PortfolioSectionProps = {
  Icon: IconComponent;
  title: string;
  children: ReactNode;
};

export const PortfolioSection: FC<PortfolioSectionProps> = ({
  Icon,
  title,
  children,
}) => {
  return (
    <div
      className="my-8 rounded-2xl border border-zinc-100 p-6 dark:border-zinc-700/40"
      data-testid="PortfolioSection"
    >
      <h2 className="flex text-sm font-semibold text-zinc-900 dark:text-zinc-100">
        <Icon className="h-6 w-6 flex-none" />
        <span className="ml-3">{title}</span>
      </h2>
      {children}
    </div>
  );
};
