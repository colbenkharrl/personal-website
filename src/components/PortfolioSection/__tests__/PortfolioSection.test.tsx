import { render, screen } from '@testing-library/react';
import * as stories from '../__stories__/PortfolioSection.stories';
import { composeStories } from '@storybook/react';

const { Basic } = composeStories(stories);

it('PortfolioSection matches snapshot (regression test)', () => {
  render(<Basic />);
  expect(screen.getByTestId('PortfolioSection')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
