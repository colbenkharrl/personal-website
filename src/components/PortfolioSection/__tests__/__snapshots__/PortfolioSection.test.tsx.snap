// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioSection matches snapshot (regression test): Full component snapshot. 1`] = `
<div
  class="my-8 rounded-2xl border border-zinc-100 p-6 dark:border-zinc-700/40"
  data-testid="PortfolioSection"
>
  <h2
    class="flex text-sm font-semibold text-zinc-900 dark:text-zinc-100"
  >
    <svg
      aria-hidden="true"
      class="h-6 w-6 flex-none"
      fill="none"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
      viewBox="0 0 24 24"
    >
      <path
        class="fill-zinc-100 stroke-zinc-400 dark:fill-zinc-100/10 dark:stroke-zinc-500"
        d="M2.75 9.75a3 3 0 0 1 3-3h12.5a3 3 0 0 1 3 3v8.5a3 3 0 0 1-3 3H5.75a3 3 0 0 1-3-3v-8.5Z"
      />
      <path
        class="stroke-zinc-400 dark:stroke-zinc-500"
        d="M3 14.25h6.249c.484 0 .952-.002 1.316.319l.777.682a.996.996 0 0 0 1.316 0l.777-.682c.364-.32.832-.319 1.316-.319H21M8.75 6.5V4.75a2 2 0 0 1 2-2h2.5a2 2 0 0 1 2 2V6.5"
      />
    </svg>
    <span
      class="ml-3"
    >
      Portfolio Section
    </span>
  </h2>
  <h1
    class="m-2"
  >
    Here is some content.
  </h1>
</div>
`;
